homeassistant/components/nexia/__init__.py

```diff
@@ -66,8 +68,23 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
-    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-    if unload_ok:
+    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
         hass.data[DOMAIN].pop(entry.entry_id)
-
     return unload_ok
+
+
+async def async_remove_config_entry_device(
+    hass: HomeAssistant, config_entry: ConfigEntry, device_entry: dr.DeviceEntry
+) -> bool:
+    """Remove a nexia config entry from a device."""
+    coordinator: NexiaDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    nexia_home: NexiaHome = coordinator.nexia_home
+    all_identifiers: set[tuple[str, str]] = {
+        (DOMAIN, automation_id) for automation_id in nexia_home.get_automation_ids()
+    }
+    for thermostat_id in nexia_home.get_thermostat_ids():
+        all_identifiers.add((DOMAIN, thermostat_id))
```

Comment:
Instead of building up all identifiers, couldn't you turn it around. First extract the nexonia ID from the device to be removed, then compare it to `thermostat_id` or `zone_id` ?